.github/workflows/release-build-publish.yml

```diff
@@ -0,0 +1,256 @@
+    permissions:
+        id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+      - name: Fetch build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: package
+          path: dist/
+
+      - name: Publish to TestPyPI
+        uses: pypa/gh-action-pypi-publish@e53eb8b103ffcb59469888563dc324e3c8ba6f06
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          verify-metadata: true
+
+  verify-version:
+    needs: [publish-testpypi, store_published_version, extract_project_name]
+    runs-on: ubuntu-latest
+    environment: testpypi
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install package from TestPyPI
+        env:
+          CLEANLAB_PACKAGE_NAME: ${{ needs.extract_project_name.outputs.cleanlab_package_name }}
+        run: pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ $CLEANLAB_PACKAGE_NAME
+
+      - name: Verify the published version
+        env:
+          VERSION: ${{ needs.store_published_version.outputs.version }}
+        run: |
+          TEST_IMPORT_VERSION=$(python -c "import cleanlab; print(cleanlab.__version__)")
+          if [ "$TEST_IMPORT_VERSION" != "$VERSION" ]; then
+            echo "Imported version $TEST_IMPORT_VERSION does not match tag $VERSION"
+            exit 1
+          fi
+
+  test-basic-import:
+    needs: [verify-version, extract_project_name]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install package from TestPyPI
+        env:
+          CLEANLAB_PACKAGE_NAME: ${{ needs.extract_project_name.outputs.cleanlab_package_name }}
+        run: pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ $CLEANLAB_PACKAGE_NAME
+
+      - name: Test importing package
+        run: |
+          python -c "import cleanlab
+          from cleanlab.outlier import OutOfDistribution
+          ood = OutOfDistribution()
+          print(ood)"
+
+  test-extras-import:
+    needs: [verify-version, extract_project_name]
+    runs-on: ubuntu-latest
+ 
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install package with extras
+        env:
+          CLEANLAB_PACKAGE_NAME: ${{ needs.extract_project_name.outputs.cleanlab_package_name }}
+        run: pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ $CLEANLAB_PACKAGE_NAME[all]
+
+      - name: Test importing package with extras
+        run: |
+          python -c "import cleanlab
+          from cleanlab import Datalab
+          lab = Datalab({'a': [1, 2, 3, 4, 5], 'b': ['a', 'b', 'c', 'b', 'a']})
+          print(lab)"
```

Comment:
Similar to the point in [this comment](https://github.com/cleanlab/cleanlab/pull/1036/files#r1513401883), no unit tests are run. But this tests that we can import Datalab functionality when we have the datalab extra installed.